### PR TITLE
[FLINK-5432] Fix nested files enumeration in ContinuousFileMonitoringFunction

### PR DESCRIPTION
**Pull request by ymarzougui**
This PR fixes reading nested files when the InputFormat has NestedFileEnumeration set to true. Nested files were not read because the code in `listEligibleFiles` did not recursively enumerate the input paths.

### CODE REVIEW

**Comment by aljoscha** [comment]
The changes look very good! I think it would be good to add a test for nested reading in `ContinuousFileProcessingTest`.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ private void monitorDirAndForwardSplits(FileSystem fs,
 	 * Returns the paths of the files not yet processed.
 	 * @param fileSystem The filesystem where the monitored directory resides.
 	 */
-	private Map<Path, FileStatus> listEligibleFiles(FileSystem fileSystem) throws IOException {
+	private Map<Path, FileStatus> listEligibleFiles(FileSystem fileSystem, String path) throws IOException {
```
Comment (zentol) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java]: I would suggest passing a `Path` here. It is always a safer option to rely on the this class than on strings.

**Comment by ymarzougui** [comment]
@aljoscha Added a test and changed Path type according to @zentol's suugestion.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java (hunk 1 above)
Comment (ymarzougui) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java]: Thanks, changed in the last commit.

**Comment by aljoscha** [comment]
@zentol do you want to recheck and merge if good?

**Comment by zentol** [comment]
looks good to me, adding it to my next batch,

**Pull request closed without merging**